src/core/ddsi/src/ddsi_discovery_topic.c

```diff
@@ -41,10 +42,14 @@ static int ddsi_sedp_write_topic_impl (struct ddsi_writer *wr, int alive, const
   ps.topic_guid = *guid;
 
   assert (xqos != NULL);
-  ps.present |= PP_PROTOCOL_VERSION | PP_VENDORID;
```

Comment:
Like with the endpoints, I wouldn't add it for the topics either. I think we should just clean up/improve version information in SPDP.